app/core/users/endpoints_users.py

```diff
@@ -843,6 +843,45 @@ async def update_current_user(
     await cruds_users.update_user(db=db, user_id=user.id, user_update=user_update)
 
 
+@router.patch(
+    "/users/external",
+    status_code=204,
+)
+async def switch_external_user_internal(
+    db: AsyncSession = Depends(get_db),
+    u: models_core.CoreUser = Depends(is_user_a_member_of(GroupType.admin)),
+):
+    """
+    Switch all external users to internal users if they have an ECL email address.
+
+    **This endpoint is only usable by administrators**
+    """
+
+    users = await cruds_users.get_users(db=db)
+    for user in users:
+        if re.match(r"^[\w\-.]*@etu(-enise)?\.ec-lyon\.fr$", user.email):
```

Comment:
Could we create a global variable with the regex in this file?

Reply:
No